### ISSUE TITLE
Inconsistent handling of missing dd in dl

### ISSUE THREAD

**Issue by FinzelStefan** [validator/validator#846]
Anerror is reported only for the last missing child element dd in dl and it is reported only for the last dt. See example:

[dl.txt](https://github.com/validator/validator/files/3446543/dl.txt)

**Comment by sideshowbarker** [comment]
The checker is behaving correctly here; see the content model for the `dl` element in the spec:

https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element

> Either: Zero or more groups each consisting of one or more `dt` elements followed by one or more `dd` elements, optionally intermixed with script-supporting elements.

```html
<dl>
<dt>dt1</dt>
<dt>dt2</dt>
<dd>dd2</dd>
<dt>dt3</dt>
</dl>
```

So you have one or more `dt` elements — `<dt>dt1</dt>` and `<dt>dt2</dt>` — followed by a `dd` element. But then after that you have a `dt` —  `<dt>dt3</dt>` that’s not followed by a `dd`.

So the checker is reporting the error in the correct place.

In other words, `<dt>foo</dt>bar<dt></dt><dd>baz</dd>` is valid per the HTML — you can have multiple `dt` elements associated with a single `dd`; the first `<dt>foo</dt>` isn’t required to have a separate `dd` element.
